node/src/components/chainspec_loader/chainspec.rs

```diff
@@ -130,25 +130,21 @@ impl Loadable for Vec<GenesisAccount> {
     fn from_file<P: AsRef<Path>>(path: P) -> Result<Self, Self::Error> {
         #[derive(Debug, Deserialize)]
         struct ParsedAccount {
-            public_key: String,
-            algorithm: String,
-            balance: String,
-            bonded_amount: String,
+            public_key: PublicKey,
+            balance: U512,
+            bonded_amount: U512,
```

Comment:
Should this also have a u8 for algorithm?  This looks like it lost the algorithm.

Reply:
I _believe_ from the changes to accounts.csv this is intended to be derived from the key itself now - as the type is prefixed onto the key (the additional `01` at the start of each). `algorithm` would then be ignored by serde.

Reply:
The algorithm is identified by the first two characters of the hex bytes (you'll see the accounts.csv files have been updated to include that info).  So the `PublicKey` type is an enum which can be parsed directly from a hex-encoded string.